optional.h

```diff
@@ -0,0 +1,31 @@
+#include <cassert>
+#include <iostream>
+#include <string>
+
+template <class T>
+class Optional {
+public:
+	Optional() = default;
+	Optional(const T& t) :
+		m_value(t),
+		m_valid(true)
+	{ }
+	operator bool() const {
+		return m_valid;
+	}
+
+	T& get() {
```

Comment:
So, what would happen if the user attempts to access "empty" optional (built in release mode)? What is the desired behavior?

Reply:
In release mode in my implementation he will have an empty object.
I could also throw an exception.
But I still have the feeling that I am doing something wrong here, since if T is a big object even when empty, we (I) allocate it on the stack even for non valid Optional.

Reply:
Please, read the assignment thoroughly.